components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -3330,15 +3330,18 @@ public static boolean jsFunction_updateApplication(Context cx,
            if (apps == null || apps.length == 0) {
                 return false;
             }
-            //check whether there is an app with same name
+
+            Map appsMap = new HashMap();
             for (Application app : apps) {
-                if (app.getName().equals(name)) {
-                    return false;
-                }
+                appsMap.put(app.getName(), app);
             }
 
             for (Application app : apps) {
-                if (app.getName().equals(oldName)) {
+                if(appsMap.containsKey(app.getName())){
```

Comment:
We seem to be iterating the same list twice. In the first iteration we're adding all items to a map and then in the second iteration we're checking if those items exists. Since we're adding all items in the first iteration (appsMap.put(app.getName(),..), they would definitely be available when we're doing the second iteration (if(appsMap.containsKey(app.getName()))).